File: web/app/Models/Item.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\MorphMany;
use Spatie\MediaLibrary\HasMedia;
use Spatie\MediaLibrary\InteractsWithMedia;

class Item extends Model implements HasMedia
{
    use HasFactory, InteractsWithMedia;

    protected $fillable = [
        'code',
        'name',
        'description',
        'gender',
        'purchase_price',
        'sale_price',
        'vat',
        'image',
        'barcode',
        'qrcode',
        'min_stock_quantity',
        'last_reorder_date',
        'supplier_id',
        'serial_number',
    ];

    protected $hidden = [
        'supplier_id',
        'updated_at',
        'created_at',
    ];

    protected $casts = [
        'last_reorder_date' => 'date',
    ];

    /*
    |--------------------------------------------------------------------------
    | Relationships
    |--------------------------------------------------------------------------
    */
    public function supplier(): BelongsTo
    {
        return $this->belongsTo(Supplier::class);
    }

    public function stocks(): HasMany
    {
        return $this->hasMany(Stock::class);
    }

    public function categories(): BelongsToMany
    {
        return $this->belongsToMany(Category::class, 'item_categories');
    }

    public function purchaseItems(): HasMany
    {
        return $this->hasMany(PurchaseItem::class);
    }

    public function saleItems(): HasMany
    {
        return $this->hasMany(SaleItem::class);
    }



}

File: mobile/lib/features/items/data/models/item.dart
import 'package:warehouse_manager/core/models/has_form.dart';
import 'package:warehouse_manager/core/models/serializable.dart';
import '../../../suppliers/data/models/supplier.dart';
import 'category.dart';

class Item implements Serializable, HasForm {

  final int id;
  final String code;
  final String name;
  final String description;
  final String gender;
  final double purchasePrice;
  final double salePrice;
  final double vat;
  final int minStockQuantity;
  DateTime lastReorderDate = DateTime.now();
  final String serialNumber;
  Supplier supplier = Supplier();
  final List<Category> categories;

  Item({
    this.id = 0,
    this.code = "",
    this.name = "",
    this.description = "",
    this.gender = "",
    this.purchasePrice = 0.0,
    this.salePrice = 0.0,
    this.vat = 0.0,
    this.minStockQuantity = 0,
    this.serialNumber = "",
    //this.supplier = const Supplier(),
    this.categories = const [],
  });

  factory Item.fromJson(Map<String, dynamic> json) {
    return Item(
      id: json['id'] ?? 0,
      code: json['code'],
      name: json['name'],
      description: json['description'],
      gender: json['gender'],
      purchasePrice: json['purchase_price'],
      salePrice: json['sale_price'],
      vat: double.parse(json['vat']),
      minStockQuantity: json['min_stock_quantity'],
     // lastReorderDate: DateTime.parse(json['last_reorder_date']),
      serialNumber: json['serial_number'],
     // supplier: Supplier.fromJson(json['supplier']),
     //  categories: (json['categories'] as List)
     //      .map((category) => Category.fromJson(category))
     //      .toList(),
    );
  }

  @override
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'code': code,
      'name': name,
      'description': description,
      'gender': gender,
      'purchase_price': purchasePrice,
      'sale_price': salePrice,
      'vat': vat,
      'min_stock_quantity': minStockQuantity,
      'last_reorder_date': lastReorderDate.toString(),
      'serial_number': serialNumber,
     // 'supplier': supplier.toJson(),
    //  'categories': categories.map((category) => category.toJson()).toList(),
    };
  }

  @override
  Map<String, dynamic> toFormData() {
    return {
      'id': id.toString(),
      'code': code,
      'name': name,
      'description': description
    };
  }

  @override
  factory Item.fromFormData(Map<String, dynamic> formData) {
    return Item(
      id: 0,
      code: formData['code'],
      name: formData['name'],
      description: formData['description'],
    );
  }
}
